controllers/ibmpowervscluster_controller.go

```diff
@@ -330,20 +330,18 @@ func (r *IBMPowerVSClusterReconciler) reconcile(clusterScope *scope.PowerVSClust
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
-	// update cluster object with loadbalancer host
-	loadBalancer := clusterScope.PublicLoadBalancer()
-	if loadBalancer == nil {
-		return reconcile.Result{}, fmt.Errorf("failed to fetch public loadbalancer")
-	}
-
 	clusterScope.Info("Getting load balancer host")
-	hostName := clusterScope.GetLoadBalancerHostName(loadBalancer.Name)
+	hostName, err := clusterScope.GetPublicLoadBalancerHostName()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to fetch public loadbalancer: %w", err)
+	}
 	if hostName == nil || *hostName == "" {
```

Comment:
> Currently we expect at least one public loadbalancer right, Otherwise we will endup in infinite loop here

With the current flow, this might not occur but added a check just to be safe.